app/controllers/admin/categories_controller.rb

```diff
@@ -57,23 +57,8 @@ def update
   end
 
   def order
-    sort_priorities = params[:order]
-                      .reject { |o| !o.match /[0-9]+/} #Guard against sql injection
```

Comment:
`.reject { |o| !o.match /[0-9]+/} #Guard against sql injection`

This regex allowed potential SQL injections to pass the `reject`. Correct regex in Ruby would be `/\A[0-9]+\z/` to match the string to contain only numbers.

The category id's are casted to integers with `to_i` later though, in the original code, so no injection _should_ actually be possible. 

I refactored this mainly for clarity and readability.